docs/arch/adr-010-new-versioning-scheme.md

```diff
@@ -0,0 +1,235 @@
+
+```yaml
+apiVersion: "cluster.kurl.sh/v1beta2"
+kind: "Installer"
+metadata:
+  name: "cluster"
+spec:
+  kubernetes:  "1.27"
+  installerVersion: "v2023.04.24-0"
+```
+
+#### Add-ons and their respective Installer Versions
+
+In order to maintain greater control over the installation process, we have also decided to modify our approach to add-on selection. Going forward, we will no longer allow users to freely select any add-on on any version of their choosing. Instead, we will provide users with specific groups of add-ons.
+
+Add-on groups are directly related to _Installer Versions_ e _Kubernetes Minor Versions_, meaning that each _Installer Version_ in a given _Kubernetes Minor Version_ will include its own specific group of add-ons. This is a hypothetical example of an add-ons group for `installerVersion` version  `v2023.04.24-0` , _Kubernetes Minor Version_ `v1.24`:
+
+| Add-on     | Version              |
+|------------|----------------------|
+| Containerd | 1.6.19               |
+| Flannel    | 0.21.3               |
+| OpenEBS    | 3.4.0                |
+| Minio      | 2023-03-13T19-46-17Z |
+
+As we release new _Installer Versions_, the add-ons included in each one of them may be updated to reflect the latest features and functionalities. This, for example, could be the add-ons grouped under `installerVersion` version `v2023.04.25-0`  on _Kubernetes Minor Version_ `v1.24` (_versions for Containerd and Flannel were updated._):
+
+| Add-on     | Version              |
+|------------|----------------------|
+| Containerd | 1.6.20               |
+| Flannel    | 0.21.4               |
+| OpenEBS    | 3.4.0                |
+| Minio      | 2023-03-13T19-46-17Z |
+
+_If no `installerVersion` parameter is specified in the Installer, the default behaviour is to install or update using the most up-to-dated Installer Version available._
+
+In summary, users can select which add-ons and their respective versions they want by pinning a specific `installerVersion` and a specific _Kubernetes Minor Versions_ in the Installer YAML.
+
+### Additional Details
+
+We will develop an approach that does not require any modifications to the kURL installer otherwise pinning old installer versions will result in a failure. Instead, we will implement a new feature in the kURL-API that allows us to seamlessly translate the new Installer (_Kubernetes Minor Version_ and _Installer Version_) into the old Installer object on-the-fly, and then inject the result into the installer script.
+
+For example, let's supposed that this is the most up-to-date add-on group for `installerVersion` version `v2023.04.24-0` in the _Kubernetes Minor Version_ 1.24:
+
+| Add-on     | Version              |
+|------------|----------------------|
+| Containerd | 1.6.20               |
+| Flannel    | 0.21.4               |
+| OpenEBS    | 3.4.0                |
+| Minio      | 2023-03-13T19-46-17Z |
```

Comment:
Minio is Addon that has many releases, so I understand that we would still upgrading this one for example per release. Am I right? Ditto so, should we not have the bundle description on the release notes of each k8s version supported ? 

What do you think ?